grails-app/services/org/bbop/apollo/FeatureService.groovy

```diff
@@ -2468,4 +2472,339 @@ public void setTranslationEnd(Transcript transcript, int translationEnd) {
+        currentFeatureJsonObject.remove(FeatureStringEnum.OWNER.value.toLowerCase())
+        currentFeatureJsonObject.remove(FeatureStringEnum.DATE_CREATION.value)
+        currentFeatureJsonObject.remove(FeatureStringEnum.DATE_LAST_MODIFIED.value)
+        if (currentFeatureJsonObject.has(FeatureStringEnum.CHILDREN.value)) {
+            for(JSONObject childFeature : currentFeatureJsonObject.get(FeatureStringEnum.CHILDREN.value)) {
+                childFeature.remove(FeatureStringEnum.ID.value)
+                childFeature.remove(FeatureStringEnum.OWNER.value.toLowerCase())
+                childFeature.remove(FeatureStringEnum.DATE_CREATION.value)
+                childFeature.remove(FeatureStringEnum.DATE_LAST_MODIFIED.value)
+                childFeature.get(FeatureStringEnum.PARENT_TYPE.value).name = type
+            }
+        }
+
+
+        if (!singletonFeatureTypes.contains(originalType) && rnaFeatureTypes.contains(type)) {
+            // *RNA to *RNA
+            if (transcriptList.size() == 1) {
+                featureRelationshipService.deleteFeatureAndChildren(parentGene)
+            }
+            else {
+                featureRelationshipService.removeFeatureRelationship(parentGene, feature)
+                featureRelationshipService.deleteFeatureAndChildren(feature)
+            }
+
+            log.debug "Converting ${originalType} to ${type}"
+
+            Transcript transcript = null
+            if (type == MRNA.alternateCvTerm) {
+                // *RNA to mRNA
+                transcript = generateTranscript(currentFeatureJsonObject, sequence, true)
+                setLongestORF(transcript)
+            }
+            else {
+                // *RNA to *RNA
+                transcript = addFeature(currentFeatureJsonObject, sequence, user, true)
+                setLongestORF(transcript)
+            }
+
+            Gene newGene = transcriptService.getGene(transcript)
+            newGene.symbol = parentGeneSymbol
+            newGene.description = parentGeneDescription
+
+            parentGeneDbxrefs.each { it ->
+                DBXref dbxref = new DBXref(
+                        db: it.db,
+                        accession: it.accession,
+                        version: it.version,
+                        description: it.description
+                ).save(flush: true)
+                newGene.addToFeatureDBXrefs(dbxref)
+            }
+
+            parentGeneFeatureProperties.each { it ->
+                if (it instanceof Comment) {
+                    featurePropertyService.addComment(newGene, it.value)
+                }
+                else {
+                    FeatureProperty fp = new FeatureProperty(
+                            type: it.type,
+                            value: it.value,
+                            rank: it.rank,
+                            tag: it.tag,
+                            feature: newGene
+                    ).save(flush: true)
```

Comment:
same comment as above . . . best not to flush in a loop if you are doing a final fush